### PR TITLE
Update to PDK 2.7

### PR DESCRIPTION
**Pull request by kreczko**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -1,291 +1,332 @@
...
+  String $template_ganglia               = $htcondor::params::template_ganglia,
+  String $template_workernode            = $htcondor::params::template_workernode,
+  String $template_defrag                = $htcondor::params::template_defrag,
+  String $template_sharedport            = $htcondor::params::template_sharedport,
+  String $template_logging               = $htcondor::params::template_logging,
+  String $template_custom_knobs          = $htcondor::params::template_custom_knobs,
+  String $template_singularity           = $htcondor::params::template_singularity,
+  String $template_highavailability      =
+    $htcondor::params::template_highavailability,
+  Boolan $use_htcondor_account_mapping   =
```
Comment (olifre) [inline manifests/init.pp:270]: `Boolan` => `Boolean`

File: manifests/init.pp [hunk 2]
```diff
@@ -1,291 +1,332 @@
...
+    $htcondor::params::use_htcondor_account_mapping,
+  Array $queue_super_users              = $htcondor::params::queue_super_users,
+  String $queue_super_user_impersonate   = $htcondor::params::queue_super_user_impersonate,
+  Boolan $use_anonymous_auth             = $htcondor::params::use_anonymous_auth,
```
Comment (olifre) [inline manifests/init.pp:274]: Same here, probably search/replace is easiest :wink:.

File: manifests/repositories.pp [hunk 3]
```diff
@@ -10,9 +10,9 @@
   $apt_key_id      = $htcondor::apt_key_id
   $apt_key_source  = $htcondor::apt_key_source
   $condor_priority = $htcondor::condor_priority
-  $major_release   = regsubst($::operatingsystemrelease, '^(\d+)\.\d+$', '\1')
+  $major_release   = regsubst($facts['os']['name'], '^(\d+)\.\d+$', '\1')
```
Comment (olifre) [inline manifests/repositories.pp:13]: I think that should be `$facts['os']['release']['full']` or maybe drop the `regsubst` and use `$facts['os']['release']['major']`?

**Review by olifre — COMMENTED** [review]
I found only some smaller things — it's interesting PDK did not spot the `Boolan` typo. 
Great work! :+1:

**Comment by kreczko** [comment]
@olifre Thank you for checking!
`pdk` and `puppet-lint` not checking the types is weird - could not find the reason why.

**Pull request merged**